src/platform/stm32f1/spi.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * SPDX-License-Identifier: MIT
+ * SPDX-FileCopyrightText: 2021 Rafael Silva <silvagracarafael@gmail.com>
+ */
+
+#include "platform/stm32f1/spi.h"
+#include "platform/stm32f1/gpio.h"
+#include "platform/stm32f1/rcc.h"
+#include "util/types.h"
+
+int spi_init(
+	struct spi_drv_data_t *drv_data,
+	u8 interface_no,
+	enum spi_mode mode,
+	u32 frequency,
+	u8 bit_order,
+	struct gpio_pin_t cs_gpio,
+	u8 cs_inverted)
+{
+	struct rcc_clock_tree_t clock_tree = rcc_get_clock_tree();
+	u32 src_clck;
+
+	switch (interface_no) {
```

Comment:
Maybe we could also make the interface an enum, and put the `#if defined(SPI*)` there. This would make the compiler blow up if the interface is not available instead of erroring out at runtime. Does that make sense?